scripts/coverage.sh

```diff
@@ -1,136 +1,69 @@
-export CARGO_INCREMENTAL=0
-export RUST_BACKTRACE=1
-export RUST_MIN_STACK=8388608
-
-echo "--- remove old coverage results"
-if [[ -d target/cov ]]; then
-  find target/cov -type f -name '*.gcda' -delete
+  PACKAGES=("$@")
 fi
-rm -rf target/cov/$reportName
-mkdir -p target/cov
-
-# Mark the base time for a clean room dir
-touch target/cov/before-test
-
-# Force rebuild of possibly-cached proc macro crates and build.rs because
-# we always want stable coverage for them
-# Don't support odd file names in our repo ever
-if [[ -n $CI || -z $1 ]]; then
-  # shellcheck disable=SC2046
-  touch \
-    $(git ls-files :**/build.rs) \
-    $(git grep -l "proc-macro.*true" :**/Cargo.toml | sed 's|Cargo.toml|src/lib.rs|')
-fi
-
-#shellcheck source=ci/common/limit-threads.sh
-source ci/common/limit-threads.sh
-
-_ "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov --no-run "${packages[@]}"
 
 TEST_ARGS=(
-  --nocapture
   --skip shred::merkle::test::test_make_shreds_from_data::
   --skip shred::merkle::test::test_make_shreds_from_data_rand::
   --skip shred::merkle::test::test_recover_merkle_shreds::
 )
 
 # most verbose log level (trace) is enabled for all solana code to make log!
 # macro code green always
-if RUST_LOG=solana=trace _ ci/intercept.sh "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov "${packages[@]}" -- "${TEST_ARGS[@]}" ; then
-  test_status=0
-else
-  test_status=$?
-  echo "Failed: $test_status"
-  echo "^^^ +++"
-  if [[ -n $CI ]]; then
-    exit $test_status
-  fi
-fi
-touch target/cov/after-test
+RUST_LOG="solana=trace agave=trace $RUST_LOG" ci/intercept.sh cargo +"$rust_nightly" test "${PACKAGES[@]}" -- "${TEST_ARGS[@]}"
 
+# Generate test reports
 echo "--- grcov"
-
-# Create a clean room dir only with updated gcda/gcno files for this run,
-# because our cached target dir is full of other builds' coverage files
-rm -rf target/cov/tmp
-mkdir -p target/cov/tmp
-
-# Can't use a simpler construct under the condition of SC2044 and bash 3
-# (macOS's default). See: https://github.com/koalaman/shellcheck/wiki/SC2044
-find target/cov -type f -name '*.gcda' -newer target/cov/before-test ! -newer target/cov/after-test -print0 |
-  (while IFS= read -r -d '' gcda_file; do
-    gcno_file="${gcda_file%.gcda}.gcno"
-    ln -sf "../../../$gcda_file" "target/cov/tmp/$(basename "$gcda_file")"
-    ln -sf "../../../$gcno_file" "target/cov/tmp/$(basename "$gcno_file")"
-  done)
-
-(
-  grcov_args=(
-    target/cov/tmp
-    --llvm
-    --ignore \*.cargo\*
-    --ignore \*build.rs
-    --ignore bench-tps\*
-    --ignore upload-perf\*
-    --ignore bench-streamer\*
-    --ignore local-cluster\*
-  )
-
-  set -x
-  grcov "${grcov_args[@]}" -t html -o target/cov/$reportName
-  grcov "${grcov_args[@]}" -t lcov -o target/cov/lcov.info
-
-  cd target/cov
-  tar zcf report.tar.gz $reportName
+grcov_common_args=(
+  "$here/.."
+  --source-dir "$here/.."
+  --binary-path "$here/../target/debug/"
```

Comment:
(this needs to be updated with target/cov, if you agree about keeping this old behavior)